### PR TITLE
Use ActsAsStiLeafClass and move Mixins into base classes

### PR DESCRIPTION
**Pull request by agrare**
Builds on ~~https://github.com/ManageIQ/manageiq-providers-kubernetes/pull/408~~ https://github.com/ManageIQ/manageiq-providers-kubernetes/pull/410 and moves mixins into base classes so they can be inherited by the Openshift provider

Depends:
- [ ] https://github.com/ManageIQ/manageiq/pull/20756
- [x] https://github.com/ManageIQ/manageiq/pull/20801

Dependents:
- [ ] Openshift: https://github.com/ManageIQ/manageiq-providers-openshift/pull/197
- [ ] Kubevirt: https://github.com/ManageIQ/manageiq-providers-kubevirt/pull/178

### CODE REVIEW

**Comment by agrare** [comment]
@miq-bot cross-repo-tests https://github.com/ManageIQ/manageiq-providers-kubernetes/pull/410, https://github.com/ManageIQ/manageiq-providers-openshift/pull/197, https://github.com/ManageIQ/manageiq-providers-kubevirt/pull/178 including https://github.com/ManageIQ/manageiq-providers-kubernetes/pull/410, https://github.com/ManageIQ/manageiq-providers-openshift/pull/197, https://github.com/ManageIQ/manageiq-providers-kubevirt/pull/178, https://github.com/ManageIQ/manageiq/pull/20756

**Comment by Fryguy** [comment]
Des this now depend on https://github.com/ManageIQ/manageiq/pull/20801 ?

**Comment by agrare** [comment]
Yes, I was hoping to get that in independently but until we subclass e.g. `Openshift::ContainerManager::ContainerNode` we still need to hard-code the type in the k8s parser

File: app/models/manageiq/providers/kubernetes/monitoring_manager.rb [hunk 1]
```diff
@@ -2,13 +2,29 @@ module ManageIQ::Providers
...
+            :authentication_status_ok,
+            :authentication_token,
+            :authentications,
+            :endpoints,
+            :zone,
+            :to        => :parent_manager,
+            :allow_nil => true
+
+    def self.hostname_required?
+     false
```
Comment (Fryguy) [inline app/models/manageiq/providers/kubernetes/monitoring_manager.rb:25]: 2 spaces?

File: app/models/manageiq/providers/kubernetes/container_manager.rb [hunk 2]
```diff
@@ -84,4 +88,1068 @@ def self.entities_for_label_mapping
...
+        },
+      ],
+    }
+  end
+
+  def self.verify_credentials(args)
+    endpoint_name = args.dig("endpoints").keys.first
+    endpoint = args.dig("endpoints", endpoint_name)
+
+    token = args.dig("authentications", "bearer", "auth_key") || args.dig("authentications", "kubevirt", "auth_key")
```
Comment (chessbyte) [inline app/models/manageiq/providers/kubernetes/container_manager.rb:757]: I know that this was not changed in this PR, but it seems strange that we are not using `endpoint_name` to decide which args to get - and then use `endpoint_name` later to make the connection.

**Review by chessbyte — APPROVED** [review]

**Pull request merged**